src/validation.h

```diff
@@ -915,9 +928,9 @@ class ChainstateManager
 
     //! The most-work chain.
     CChainState& ActiveChainstate() const;
-    CChain& ActiveChain() const { return ActiveChainstate().m_chain; }
-    int ActiveHeight() const { return ActiveChain().Height(); }
-    CBlockIndex* ActiveTip() const { return ActiveChain().Tip(); }
+    CChain& ActiveChain() const EXCLUSIVE_LOCKS_REQUIRED(GetMutex()) { return ActiveChainstate().m_chain; }
+    int ActiveHeight() const EXCLUSIVE_LOCKS_REQUIRED(GetMutex()) { return ActiveChain().Height(); }
```

Comment:
Needs `{ LOCK(::cs_main); ... }` added to bitcoin-chainstate.cpp around the `Main program logic starts here` section.

Reply:
Good find. Thanks and fixed.